### DISCUSSION TITLE
Does Input support inline custom components?

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [iTwin/iTwinUI-react#976] [Q&A]
You can add a custom button to an input component, but it appears that the layout is messed up if using any custom component. Is this intentional? Is there a way to support it?

https://codesandbox.io/s/itwinui-react-minimal-example-forked-e6ecit?file=/src/App.tsx

![image](https://user-images.githubusercontent.com/15036346/206213513-68586271-285a-4e9f-90b8-7d57164ed298.png)

**Comment by HaveSpacesuit** [comment]
It looks like using non-jsx gets this to work as I want:
```
<LabeledInput
  placeholder="Enter text here..."
  label="This is a label"
  svgIcon={MyButton()}
  iconDisplayStyle="inline"
/>
```

**Comment by mayank99** [comment]
The problem here is that your custom component does not take any props. It will work if you forward the props.
```jsx
function MyButton(props: IconButtonProps) {
  return (
    <IconButton styleType="borderless" {...props}>
      <SvgCloseSmall />
    </IconButton>
  );
}
```

This is because we clone the icon and pass it a `className`.
https://github.com/iTwin/iTwinUI-react/blob/e43d3e1c6c8c0f9ee0687f3d19c134e68bd52d4b/packages/iTwinUI-react/src/core/utils/components/InputContainer.tsx#L76-L79

In the future we might change the approach to avoid cloning (it is considered [bad practice](https://beta.reactjs.org/apis/react/cloneElement)) and instead add a wrapper. That should get rid of this limitation.

Hope that helps.
